src/app/new-auth/forgot-password/forgot-password.component.ts

```diff
@@ -1,15 +1,114 @@
+import { animate, state, style, transition, trigger } from '@angular/animations';
 
 @Component({
   selector: 'clark-forgot-password',
   templateUrl: './forgot-password.component.html',
-  styleUrls: ['./forgot-password.component.scss']
+  styleUrls: ['./forgot-password.component.scss'],
+  animations: [
+    trigger('toggleButton', [
+      state('disabled', style({
+          borderColor: 'grey',
+          backgroundColor: 'grey',
+          color: 'black',
+          opacity: 0.2
+      })),
+      state('enabled', style({
+          color: 'white',
+          opacity: 1
+      })),
+      transition('disabled <=> enabled', [
+        animate('0.2s')
+      ])
+    ]),
+    trigger('switchViews', [
+      transition(':enter', [
+        style({ opacity: 0 }),
+        animate('1000ms')
+      ])
+    ])
+  ]
 })
-export class ForgotPasswordComponent implements OnInit {
+export class ForgotPasswordComponent implements OnInit{
+  showError: Boolean = false;
+  errorMessage: String;
+  showDone: Boolean = false;
+  submitButton: String = 'disabled';
 
-  constructor() { }
+  emails: FormGroup = new FormGroup({
+    'email': this.authValidationService.getInputFormControl('email'),
+    'confirmEmail': this.authValidationService.getInputFormControl('required')
+  }, { validators: MatchValidator.mustMatch('email', 'confirmEmail') });;
+
+  emailInUse = true;
+  emailErrorMsg = '';
+  private ngUnsubscribe = new Subject<void>();
+
+  constructor(
+    private authValidationService: AuthValidationService,
+    private authService: AuthService
+  ) { }
 
   ngOnInit(): void {
+    this.authValidationService.getErrorState().subscribe(err => this.showError = err);
+    this.validateEmail();
+  }
+
+  /**
+   * Disables and grays out the 'Request Password Reset' button if there are any input errors, enables them otherwise
+   */
+  toggleButton(): void {
+    if(this.emails.controls.email.errors || this.emails.controls.confirmEmail.errors) {
+      this.submitButton = 'disabled';
+    } else {
+      this.submitButton = 'enabled';
+    }
   }
 
+  /**
+   * Sends an email to create new password to the specified email.
+   * Displays an error banner if an error is thrown.
+   */
+  submit(): void {
+    if(this.submitButton === 'enabled') {
+      this.authService.initiateResetPassword(this.emails.get('email').value).subscribe(val => {
+        this.showDone = true;
+      }, error => {
+        this.errorMessage = 'Something went wrong! We\'re looking into the issue. Please check back later.';
+        this.authValidationService.showError();
+      });
+    }
+  }
+
+  /**
+   * Validates the email by ensuring that it exists in the database
+   * Displays an error message beneath the input if the email does not exist
+   */
+  private validateEmail() {
+      this.emails.get('email').valueChanges.pipe(
+        debounce(() => {
+          return interval(600);
+        }),
+        takeUntil(this.ngUnsubscribe)).subscribe(
```

Comment:
Looks good just curious what takeUntil(this.ngUnsubscibe) is doing here, it doesn't seem to be changed anywhere.

Reply:
This component should implement ngOnDestroy and a function should be created to change `ngUnsubscribe`. Look at my PR/branch for how i did it.

Reply:
I've implemented ngOnDestroy to unsubscribe from any subscriptions. Let me know if there's something else I'm missing.